internal/jimm/cloud.go

```diff
@@ -20,35 +20,173 @@ import (
 	ofganames "github.com/canonical/jimm/v3/internal/openfga/names"
 )
 
+// GetUserCloudAccess returns users access level for the specified cloud.
+func (j *JIMM) GetUserCloudAccess(ctx context.Context, user *openfga.User, cloud names.CloudTag) (string, error) {
+	accessLevel := user.GetCloudAccess(ctx, cloud)
+	return permissions.ToCloudAccessString(accessLevel), nil
+}
+
+// GetCloud retrieves the cloud for the given cloud tag. If the cloud
+// cannot be found then an error with the code CodeNotFound is
+// returned. If the user does not have permission to view the cloud then an
+// error with a code of CodeUnauthorized is returned. If the user only has
+// add-model access to the cloud then the returned Users field will only
+// contain the authentcated user.
+func (j *JIMM) GetCloud(ctx context.Context, user *openfga.User, tag names.CloudTag) (jujuparams.Cloud, error) {
+	const op = errors.Op("jimm.CloudInfo")
+	zapctx.Info(ctx, string(op))
```

Comment:
Was thinking about this sometime last year and whether we want it. If we do want it then it might be better suited to go inside the `jujuapi.FindMethod()` class of functions so that we don't clutter up these functions.